crates/catalog/src/schema.rs

```
@@ -39,14 +40,26 @@ impl SchemaProvider for CachingSchema {
     }
 
     fn table_names(&self) -> Vec<String> {
-        if self.tables_cache.is_empty() {
-            self.schema.table_names()
-        } else {
-            // Don't fill the cache since should call async table() to fill it
-            self.tables_cache
-                .iter()
-                .map(|entry| entry.key().clone())
-                .collect()
+        match &self.iceberg_catalog {
+            Some(catalog) => {
+                let catalog = catalog.clone();
+                let Ok(namespace) = Namespace::try_new(std::slice::from_ref(&self.name)) else {
+                    return vec![];
```

Another example here - in case we failed to get a namespace, does it make sense to silently return empty table names or better we return an error "Namespace xxx is invalid" (or whatever the erorr should be?)

---

Namespace here is just a string builder, not related to catalog

---

Also we cannot return errors in this trait method :)

---

Yeah, okay, I also see function signature doesn't allow to return an error